### ISSUE TITLE
Loss function implementation

### ISSUE THREAD

**Issue by kildarj** [TinaLRJ/Multi-frame-infrared-small-target-detection-DTUM#15]
Hi Dr Li,

Thanks for sharing your code!

I just wanted to ask about the implementation of your HPM loss function. In the training script provided, you use the fullySup or fullySup1 arguments for any training, which just uses the typical soft IoU loss function:

`python train.py --model 'ResUNet_DTUM' --loss_func 'fullySup' --train 1 --test 0 --fullySupervised True
python train.py --model 'DNANet_DTUM' --loss_func 'fullySup1' --train 1 --test 0 --fullySupervised True --SpatialDeepSup False`

Only if you specify HPM rather than fullySup does the loss function you describe in your associated paper actually get applied. Are you able to clarify why this is the case?

Upon doing some testing of an implementation of your model with an alternative MIRSDT dataset, DAUB, the SoftIoU loss function outperformed the HPM loss significantly. Are you able to help explain that as well?

Kind regards,
Jordan

**Comment by TinaLRJ** [comment]
Hi Jordan,

Thank you for noting this. The HPM loss function is designed for point-supervised training. It effectively balances positive and negative samples using only point labels and helps the network focus on positive samples and hard negative samples.




Best wishes,

Ruojing Li



-----原始邮件-----
发件人:kildarj ***@***.***>
发送时间:2026-03-03 13:52:16 (星期二)
收件人: TinaLRJ/Multi-frame-infrared-small-target-detection-DTUM ***@***.***>
抄送: Subscribed ***@***.***>
主题: [TinaLRJ/Multi-frame-infrared-small-target-detection-DTUM] Loss function implementation (Issue #15)


kildarj created an issue (TinaLRJ/Multi-frame-infrared-small-target-detection-DTUM#15)

Hi Dr Li,

Thanks for sharing your code!

I just wanted to ask about the implementation of your HPM loss function. In the training script provided, you use the fullySup or fullySup1 arguments for any training, which just uses the typical soft IoU loss function:

python train.py --model 'ResUNet_DTUM' --loss_func 'fullySup' --train 1 --test 0 --fullySupervised True python train.py --model 'DNANet_DTUM' --loss_func 'fullySup1' --train 1 --test 0 --fullySupervised True --SpatialDeepSup False

Only if you specify HPM rather than fullySup does the loss function you describe in your associated paper actually get applied. Are you able to clarify why this is the case?

Upon doing some testing of an implementation of your model with an alternative MIRSDT dataset, DAUB, the SoftIoU loss function outperformed the HPM loss significantly. Are you able to help explain that as well?

Kind regards,
Jordan
